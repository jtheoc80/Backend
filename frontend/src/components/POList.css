.po-list {
  max-width: 1200px;
  margin: 0 auto;
  padding: 20px;
}

.list-header {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  margin-bottom: 30px;
  padding-bottom: 20px;
  border-bottom: 2px solid #e9ecef;
}

.header-left h2 {
  margin: 0 0 10px 0;
  color: #333;
  font-size: 24px;
  font-weight: 600;
}

.po-stats {
  display: flex;
  gap: 20px;
}

.stat {
  font-size: 14px;
  color: #6c757d;
  background: #f8f9fa;
  padding: 4px 8px;
  border-radius: 4px;
}

.list-controls {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 20px;
  gap: 20px;
}

.search-bar {
  flex: 1;
}

.search-input {
  width: 100%;
  max-width: 400px;
  padding: 10px 12px;
  border: 1px solid #ced4da;
  border-radius: 4px;
  font-size: 14px;
}

.search-input:focus {
  outline: none;
  border-color: #007bff;
  box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
}

.filters {
  display: flex;
  gap: 10px;
  align-items: center;
}

.filter-select {
  padding: 8px 10px;
  border: 1px solid #ced4da;
  border-radius: 4px;
  font-size: 14px;
  background: white;
}

.filter-select:focus {
  outline: none;
  border-color: #007bff;
}

.sort-button {
  padding: 8px 12px;
  border: 1px solid #ced4da;
  border-radius: 4px;
  background: white;
  cursor: pointer;
  font-size: 16px;
}

.sort-button:hover {
  background: #f8f9fa;
}

.po-table-container {
  background: white;
  border-radius: 8px;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  overflow: hidden;
  margin-bottom: 20px;
}

.po-table {
  width: 100%;
  border-collapse: collapse;
}

.po-table th,
.po-table td {
  text-align: left;
  padding: 12px 15px;
  border-bottom: 1px solid #e9ecef;
}

.po-table th {
  background: #f8f9fa;
  font-weight: 600;
  color: #495057;
  font-size: 14px;
  position: sticky;
  top: 0;
  z-index: 10;
}

.po-table th.sortable {
  cursor: pointer;
  user-select: none;
  transition: background-color 0.15s ease;
}

.po-table th.sortable:hover {
  background: #e9ecef;
}

.po-table tbody tr:hover {
  background: #f8f9fa;
}

.po-number {
  font-weight: 600;
}

.link-button {
  background: none;
  border: none;
  color: #007bff;
  cursor: pointer;
  text-decoration: underline;
  font-weight: 600;
  padding: 0;
}

.link-button:hover {
  color: #0056b3;
}

.amount {
  font-weight: 600;
  color: #28a745;
}

.date {
  color: #6c757d;
  font-size: 13px;
}

.status-badge {
  display: inline-block;
  padding: 4px 8px;
  border-radius: 12px;
  font-size: 12px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.5px;
}

.status-pending {
  background: #fff3cd;
  color: #856404;
  border: 1px solid #ffeaa7;
}

.status-approved {
  background: #d1edff;
  color: #155724;
  border: 1px solid #b8daff;
}

.status-rejected {
  background: #f8d7da;
  color: #721c24;
  border: 1px solid #f5c6cb;
}

.status-cancelled {
  background: #e2e3e5;
  color: #383d41;
  border: 1px solid #d6d8db;
}

.action-buttons {
  display: flex;
  gap: 5px;
}

.btn {
  padding: 6px 12px;
  border: none;
  border-radius: 4px;
  font-size: 12px;
  font-weight: 500;
  cursor: pointer;
  transition: background-color 0.15s ease-in-out;
  text-decoration: none;
  display: inline-block;
  text-align: center;
}

.btn:disabled {
  opacity: 0.65;
  cursor: not-allowed;
}

.btn-sm {
  padding: 4px 8px;
  font-size: 11px;
}

.btn-primary {
  background-color: #007bff;
  color: white;
}

.btn-primary:hover:not(:disabled) {
  background-color: #0056b3;
}

.btn-secondary {
  background-color: #6c757d;
  color: white;
}

.btn-secondary:hover:not(:disabled) {
  background-color: #545b62;
}

.btn-success {
  background-color: #28a745;
  color: white;
}

.btn-success:hover:not(:disabled) {
  background-color: #1e7e34;
}

.btn-danger {
  background-color: #dc3545;
  color: white;
}

.btn-danger:hover:not(:disabled) {
  background-color: #c82333;
}

.pagination {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 15px;
  margin-top: 20px;
}

.page-info {
  font-size: 14px;
  color: #6c757d;
}

.loading-state,
.empty-state {
  text-align: center;
  padding: 60px 20px;
  color: #6c757d;
}

.loading-spinner {
  width: 40px;
  height: 40px;
  border: 4px solid #f3f3f3;
  border-top: 4px solid #007bff;
  border-radius: 50%;
  animation: spin 1s linear infinite;
  margin: 0 auto 20px;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}

.empty-state h3 {
  margin-bottom: 10px;
  color: #495057;
}

.empty-state p {
  margin-bottom: 20px;
}

.po-list-error {
  text-align: center;
  padding: 40px 20px;
}

.error-message {
  background: #f8d7da;
  border: 1px solid #f5c6cb;
  color: #721c24;
  padding: 20px;
  border-radius: 8px;
  display: inline-block;
}

.error-message h3 {
  margin-top: 0;
  margin-bottom: 10px;
}

/* Responsive Design */
@media (max-width: 768px) {
  .po-list {
    padding: 10px;
  }
  
  .list-header {
    flex-direction: column;
    align-items: stretch;
    gap: 15px;
  }
  
  .po-stats {
    flex-wrap: wrap;
  }
  
  .list-controls {
    flex-direction: column;
    align-items: stretch;
    gap: 15px;
  }
  
  .filters {
    flex-wrap: wrap;
  }
  
  .po-table-container {
    overflow-x: auto;
  }
  
  .po-table {
    min-width: 800px;
  }
  
  .action-buttons {
    flex-direction: column;
    gap: 2px;
  }
  
  .pagination {
    flex-wrap: wrap;
    gap: 10px;
  }
}

@media (max-width: 480px) {
  .po-stats {
    flex-direction: column;
    gap: 5px;
  }
  
  .filters {
    flex-direction: column;
  }
  
  .filter-select,
  .sort-button {
    width: 100%;
  }
}