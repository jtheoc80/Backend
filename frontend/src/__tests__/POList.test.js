import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import POList from '../components/POList';

// Mock purchase orders data
const mockPurchaseOrders = [
  {
    id: 1,
    po_number: 'PO-2024-001',
    manufacturer_id: 'mfg001',
    manufacturer_name: 'Emerson Process Management',
    distributor_id: 'dist001',
    distributor_name: 'Industrial Valve Solutions Inc',
    total_amount: 1000.50,
    currency: 'USD',
    status: 'pending',
    created_at: '2024-01-15T10:30:00Z',
    updated_at: '2024-01-15T10:30:00Z'
  },
  {
    id: 2,
    po_number: 'PO-2024-002',
    manufacturer_id: 'mfg002',
    manufacturer_name: 'Kitz Corporation',
    distributor_id: 'dist002',
    distributor_name: 'Global Valve Distribution',
    total_amount: 2500.75,
    currency: 'USD',
    status: 'approved',
    created_at: '2024-01-16T14:45:00Z',
    updated_at: '2024-01-16T16:20:00Z'
  },
  {
    id: 3,
    po_number: 'PO-2024-003',
    manufacturer_id: 'mfg001',
    manufacturer_name: 'Emerson Process Management',
    distributor_id: 'dist001',
    distributor_name: 'Industrial Valve Solutions Inc',
    total_amount: 750.25,
    currency: 'EUR',
    status: 'rejected',
    created_at: '2024-01-17T09:15:00Z',
    updated_at: '2024-01-17T11:30:00Z'
  }
];

describe('POList', () => {
  const mockOnCreateNew = jest.fn();
  const mockOnViewDetails = jest.fn();
  const mockOnApprove = jest.fn();
  const mockOnReject = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  const renderComponent = (props = {}) => {
    return render(
      <POList
        onCreateNew={mockOnCreateNew}
        onViewDetails={mockOnViewDetails}
        onApprove={mockOnApprove}
        onReject={mockOnReject}
        initialPOs={mockPurchaseOrders}
        {...props}
      />
    );
  };

  describe('Component Rendering', () => {
    it('should render the main heading and create button', () => {
      renderComponent();

      expect(screen.getByText('Purchase Orders')).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /Create New PO/ })).toBeInTheDocument();
    });

    it('should display purchase order statistics', () => {
      renderComponent();

      expect(screen.getByText('Total: 3')).toBeInTheDocument();
      expect(screen.getByText('Pending: 1')).toBeInTheDocument();
      expect(screen.getByText('Approved: 1')).toBeInTheDocument();
    });

    it('should render search bar and filters', () => {
      renderComponent();

      expect(screen.getByPlaceholderText(/Search by PO number/)).toBeInTheDocument();
      expect(screen.getByDisplayValue('All Statuses')).toBeInTheDocument();
      expect(screen.getByDisplayValue('Sort by Created Date')).toBeInTheDocument();
    });

    it('should render table with all purchase orders', () => {
      renderComponent();

      expect(screen.getByText('PO-2024-001')).toBeInTheDocument();
      expect(screen.getByText('PO-2024-002')).toBeInTheDocument();
      expect(screen.getByText('PO-2024-003')).toBeInTheDocument();
      expect(screen.getAllByText('Emerson Process Management')).toHaveLength(2);
      expect(screen.getByText('Kitz Corporation')).toBeInTheDocument();
    });

    it('should display formatted currency amounts', () => {
      renderComponent();

      expect(screen.getByText('$1,000.50')).toBeInTheDocument();
      expect(screen.getByText('$2,500.75')).toBeInTheDocument();
    });

    it('should display status badges with appropriate styling', () => {
      renderComponent();

      const pendingBadge = screen.getByText((content, element) => 
        element?.classList.contains('status-pending') && content === 'Pending'
      );
      const approvedBadge = screen.getByText((content, element) => 
        element?.classList.contains('status-approved') && content === 'Approved'
      );
      const rejectedBadge = screen.getByText((content, element) => 
        element?.classList.contains('status-rejected') && content === 'Rejected'
      );

      expect(pendingBadge).toHaveClass('status-pending');
      expect(approvedBadge).toHaveClass('status-approved');
      expect(rejectedBadge).toHaveClass('status-rejected');
    });
  });

  describe('Search and Filtering', () => {
    it('should filter purchase orders by search term', async () => {
      const user = userEvent.setup();
      renderComponent();

      const searchInput = screen.getByPlaceholderText(/Search by PO number/);
      await user.type(searchInput, 'PO-2024-001');

      expect(screen.getByText('PO-2024-001')).toBeInTheDocument();
      expect(screen.queryByText('PO-2024-002')).not.toBeInTheDocument();
      expect(screen.queryByText('PO-2024-003')).not.toBeInTheDocument();
    });

    it('should filter by manufacturer name', async () => {
      const user = userEvent.setup();
      renderComponent();

      const searchInput = screen.getByPlaceholderText(/Search by PO number/);
      await user.type(searchInput, 'Kitz');

      expect(screen.getByText('PO-2024-002')).toBeInTheDocument();
      expect(screen.queryByText('PO-2024-001')).not.toBeInTheDocument();
    });

    it('should filter by status', async () => {
      const user = userEvent.setup();
      renderComponent();

      const statusFilter = screen.getByDisplayValue('All Statuses');
      await user.selectOptions(statusFilter, 'approved');

      expect(screen.getByText('PO-2024-002')).toBeInTheDocument();
      expect(screen.queryByText('PO-2024-001')).not.toBeInTheDocument();
      expect(screen.queryByText('PO-2024-003')).not.toBeInTheDocument();
    });

    it('should update statistics based on filtered results', async () => {
      const user = userEvent.setup();
      renderComponent();

      const statusFilter = screen.getByDisplayValue('All Statuses');
      await user.selectOptions(statusFilter, 'pending');

      expect(screen.getByText('Total: 1')).toBeInTheDocument();
      expect(screen.getByText('Pending: 1')).toBeInTheDocument();
      expect(screen.getByText('Approved: 0')).toBeInTheDocument();
    });
  });

  describe('Sorting', () => {
    it('should sort by PO number when header is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      const poNumberHeader = screen.getByRole('columnheader', { name: /PO Number/ });
      await user.click(poNumberHeader);

      // Check that the sort icon changes (testing the sorting functionality)
      expect(poNumberHeader).toBeInTheDocument();
    });

    it('should change sort order when same column is clicked twice', async () => {
      const user = userEvent.setup();
      renderComponent();

      const amountHeader = screen.getByRole('columnheader', { name: /Amount/ });
      await user.click(amountHeader);
      await user.click(amountHeader);

      expect(amountHeader).toBeInTheDocument();
    });

    it('should sort by different fields using dropdown', async () => {
      const user = userEvent.setup();
      renderComponent();

      const sortSelect = screen.getByDisplayValue('Sort by Created Date');
      await user.selectOptions(sortSelect, 'total_amount');

      expect(sortSelect).toHaveValue('total_amount');
    });
  });

  describe('Action Buttons', () => {
    it('should show appropriate action buttons based on status', () => {
      renderComponent();

      // For pending PO, should show Approve and Reject buttons
      const pendingRow = screen.getByText('PO-2024-001').closest('tr');
      expect(pendingRow).toHaveTextContent('Approve');
      expect(pendingRow).toHaveTextContent('Reject');

      // For approved PO, should only show View button
      const approvedRow = screen.getByText('PO-2024-002').closest('tr');
      expect(approvedRow).not.toHaveTextContent('Approve');
      expect(approvedRow).not.toHaveTextContent('Reject');
    });

    it('should call onViewDetails when View button is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      const viewButtons = screen.getAllByRole('button', { name: /View/ });
      await user.click(viewButtons[0]);

      expect(mockOnViewDetails).toHaveBeenCalledWith(mockPurchaseOrders[0]);
    });

    it('should call onViewDetails when PO number link is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      const poNumberLink = screen.getByRole('button', { name: 'PO-2024-001' });
      await user.click(poNumberLink);

      expect(mockOnViewDetails).toHaveBeenCalledWith(mockPurchaseOrders[0]);
    });

    it('should call onApprove when Approve button is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      const approveButton = screen.getByRole('button', { name: /Approve/ });
      await user.click(approveButton);

      expect(mockOnApprove).toHaveBeenCalledWith(mockPurchaseOrders[0]);
    });

    it('should call onReject when Reject button is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      const rejectButton = screen.getByRole('button', { name: /Reject/ });
      await user.click(rejectButton);

      expect(mockOnReject).toHaveBeenCalledWith(mockPurchaseOrders[0]);
    });

    it('should call onCreateNew when Create New PO button is clicked', async () => {
      const user = userEvent.setup();
      renderComponent();

      const createButton = screen.getByRole('button', { name: /Create New PO/ });
      await user.click(createButton);

      expect(mockOnCreateNew).toHaveBeenCalledTimes(1);
    });
  });

  describe('Loading State', () => {
    it('should show loading spinner when loading is true', () => {
      renderComponent({ loading: true });

      expect(screen.getByText('Loading purchase orders...')).toBeInTheDocument();
      expect(screen.getByTestId('loading-spinner')).toBeInTheDocument();
    });

    it('should disable create button when loading', () => {
      renderComponent({ loading: true });

      const createButton = screen.getByRole('button', { name: /Create New PO/ });
      expect(createButton).toBeDisabled();
    });
  });

  describe('Error State', () => {
    it('should display error message when error prop is provided', () => {
      const errorMessage = 'Failed to load purchase orders';
      renderComponent({ error: errorMessage });

      expect(screen.getByText('Error Loading Purchase Orders')).toBeInTheDocument();
      expect(screen.getByText(errorMessage)).toBeInTheDocument();
    });

    it('should not render table when there is an error', () => {
      renderComponent({ error: 'Some error' });

      expect(screen.queryByRole('table')).not.toBeInTheDocument();
    });
  });

  describe('Empty State', () => {
    it('should show empty state when no purchase orders exist', () => {
      renderComponent({ initialPOs: [] });

      expect(screen.getByText('No Purchase Orders Found')).toBeInTheDocument();
      expect(screen.getByText('Get started by creating your first purchase order.')).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /Create Your First PO/ })).toBeInTheDocument();
    });

    it('should show filtered empty state when no results match filters', async () => {
      const user = userEvent.setup();
      renderComponent();

      const searchInput = screen.getByPlaceholderText(/Search by PO number/);
      await user.type(searchInput, 'NONEXISTENT');

      expect(screen.getByText('No Purchase Orders Found')).toBeInTheDocument();
      expect(screen.getByText('No purchase orders match your current filters.')).toBeInTheDocument();
    });

    it('should call onCreateNew from empty state button', async () => {
      const user = userEvent.setup();
      renderComponent({ initialPOs: [] });

      const createButton = screen.getByRole('button', { name: /Create Your First PO/ });
      await user.click(createButton);

      expect(mockOnCreateNew).toHaveBeenCalledTimes(1);
    });
  });

  describe('Date Formatting', () => {
    it('should format dates correctly', () => {
      renderComponent();

      // Should show formatted date
      expect(screen.getByText(/Jan 15, 2024/)).toBeInTheDocument();
      expect(screen.getByText(/Jan 16, 2024/)).toBeInTheDocument();
    });
  });

  describe('Responsive Behavior', () => {
    it('should render mobile-friendly layout', () => {
      renderComponent();

      const table = screen.getByRole('table');
      expect(table).toHaveClass('po-table');
    });
  });

  describe('Accessibility', () => {
    it('should have proper table structure with headers', () => {
      renderComponent();

      expect(screen.getByRole('table')).toBeInTheDocument();
      expect(screen.getByRole('columnheader', { name: /PO Number/ })).toBeInTheDocument();
      expect(screen.getByRole('columnheader', { name: /Status/ })).toBeInTheDocument();
    });

    it('should have accessible buttons with proper labels', () => {
      renderComponent();

      const viewButtons = screen.getAllByRole('button', { name: /View/ });
      const approveButton = screen.getByRole('button', { name: /Approve/ });
      const rejectButton = screen.getByRole('button', { name: /Reject/ });

      expect(viewButtons.length).toBeGreaterThan(0);
      expect(approveButton).toBeInTheDocument();
      expect(rejectButton).toBeInTheDocument();
    });
  });

  describe('Pagination', () => {
    it('should not show pagination when totalPages is 1 or less', () => {
      renderComponent({ 
        initialPOs: mockPurchaseOrders,
        pagination: { currentPage: 1, totalPages: 1, totalCount: 3 }
      });

      expect(screen.queryByText(/Previous/)).not.toBeInTheDocument();
      expect(screen.queryByText(/Next/)).not.toBeInTheDocument();
    });
  });

  describe('Snapshot Testing', () => {
    it('should match snapshot with default props', () => {
      const { container } = renderComponent();
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with loading state', () => {
      const { container } = renderComponent({ loading: true });
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with error state', () => {
      const { container } = renderComponent({ error: 'Test error message' });
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with empty state', () => {
      const { container } = renderComponent({ initialPOs: [] });
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with filtered results', async () => {
      const user = userEvent.setup();
      const { container } = renderComponent();

      const statusFilter = screen.getByDisplayValue('All Statuses');
      await user.selectOptions(statusFilter, 'pending');

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});