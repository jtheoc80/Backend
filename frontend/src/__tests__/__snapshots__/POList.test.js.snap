// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POList Snapshot Testing should match snapshot with default props 1`] = `
<div
  class="po-list"
>
  <div
    class="list-header"
  >
    <div
      class="header-left"
    >
      <h2>
        Purchase Orders
      </h2>
      <div
        class="po-stats"
      >
        <span
          class="stat"
        >
          Total: 
          3
        </span>
        <span
          class="stat"
        >
          Pending: 
          1
        </span>
        <span
          class="stat"
        >
          Approved: 
          1
        </span>
      </div>
    </div>
    <button
      class="btn btn-primary"
    >
      Create New PO
    </button>
  </div>
  <div
    class="list-controls"
  >
    <div
      class="search-bar"
    >
      <input
        class="search-input"
        placeholder="Search by PO number, manufacturer, or distributor..."
        type="text"
        value=""
      />
    </div>
    <div
      class="filters"
    >
      <select
        class="filter-select"
      >
        <option
          value=""
        >
          All Statuses
        </option>
        <option
          value="pending"
        >
          Pending
        </option>
        <option
          value="approved"
        >
          Approved
        </option>
        <option
          value="rejected"
        >
          Rejected
        </option>
        <option
          value="cancelled"
        >
          Cancelled
        </option>
      </select>
      <select
        class="filter-select"
      >
        <option
          value="created_at"
        >
          Sort by Created Date
        </option>
        <option
          value="updated_at"
        >
          Sort by Updated Date
        </option>
        <option
          value="po_number"
        >
          Sort by PO Number
        </option>
        <option
          value="total_amount"
        >
          Sort by Amount
        </option>
        <option
          value="status"
        >
          Sort by Status
        </option>
      </select>
      <button
        class="sort-button"
        title="Sort Ascending"
      >
        ↓
      </button>
    </div>
  </div>
  <div
    class="po-table-container"
  >
    <table
      class="po-table"
    >
      <thead>
        <tr>
          <th
            class="sortable"
          >
            PO Number 
            ↕️
          </th>
          <th>
            Manufacturer
          </th>
          <th>
            Distributor
          </th>
          <th
            class="sortable"
          >
            Amount 
            ↕️
          </th>
          <th
            class="sortable"
          >
            Status 
            ↕️
          </th>
          <th
            class="sortable"
          >
            Created 
            ↓
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="po-row"
        >
          <td
            class="po-number"
          >
            <button
              class="link-button"
            >
              PO-2024-003
            </button>
          </td>
          <td>
            Emerson Process Management
          </td>
          <td>
            Industrial Valve Solutions Inc
          </td>
          <td
            class="amount"
          >
            €750.25
          </td>
          <td>
            <span
              class="status-badge status-rejected"
            >
              Rejected
            </span>
          </td>
          <td
            class="date"
          >
            Jan 17, 2024, 09:15 AM
          </td>
          <td>
            <div
              class="action-buttons"
            >
              <button
                class="btn btn-sm btn-secondary"
                title="View Details"
              >
                View
              </button>
            </div>
          </td>
        </tr>
        <tr
          class="po-row"
        >
          <td
            class="po-number"
          >
            <button
              class="link-button"
            >
              PO-2024-002
            </button>
          </td>
          <td>
            Kitz Corporation
          </td>
          <td>
            Global Valve Distribution
          </td>
          <td
            class="amount"
          >
            $2,500.75
          </td>
          <td>
            <span
              class="status-badge status-approved"
            >
              Approved
            </span>
          </td>
          <td
            class="date"
          >
            Jan 16, 2024, 02:45 PM
          </td>
          <td>
            <div
              class="action-buttons"
            >
              <button
                class="btn btn-sm btn-secondary"
                title="View Details"
              >
                View
              </button>
            </div>
          </td>
        </tr>
        <tr
          class="po-row"
        >
          <td
            class="po-number"
          >
            <button
              class="link-button"
            >
              PO-2024-001
            </button>
          </td>
          <td>
            Emerson Process Management
          </td>
          <td>
            Industrial Valve Solutions Inc
          </td>
          <td
            class="amount"
          >
            $1,000.50
          </td>
          <td>
            <span
              class="status-badge status-pending"
            >
              Pending
            </span>
          </td>
          <td
            class="date"
          >
            Jan 15, 2024, 10:30 AM
          </td>
          <td>
            <div
              class="action-buttons"
            >
              <button
                class="btn btn-sm btn-secondary"
                title="View Details"
              >
                View
              </button>
              <button
                class="btn btn-sm btn-success"
                title="Approve PO"
              >
                Approve
              </button>
              <button
                class="btn btn-sm btn-danger"
                title="Reject PO"
              >
                Reject
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`POList Snapshot Testing should match snapshot with empty state 1`] = `
<div
  class="po-list"
>
  <div
    class="list-header"
  >
    <div
      class="header-left"
    >
      <h2>
        Purchase Orders
      </h2>
      <div
        class="po-stats"
      >
        <span
          class="stat"
        >
          Total: 
          0
        </span>
        <span
          class="stat"
        >
          Pending: 
          0
        </span>
        <span
          class="stat"
        >
          Approved: 
          0
        </span>
      </div>
    </div>
    <button
      class="btn btn-primary"
    >
      Create New PO
    </button>
  </div>
  <div
    class="list-controls"
  >
    <div
      class="search-bar"
    >
      <input
        class="search-input"
        placeholder="Search by PO number, manufacturer, or distributor..."
        type="text"
        value=""
      />
    </div>
    <div
      class="filters"
    >
      <select
        class="filter-select"
      >
        <option
          value=""
        >
          All Statuses
        </option>
        <option
          value="pending"
        >
          Pending
        </option>
        <option
          value="approved"
        >
          Approved
        </option>
        <option
          value="rejected"
        >
          Rejected
        </option>
        <option
          value="cancelled"
        >
          Cancelled
        </option>
      </select>
      <select
        class="filter-select"
      >
        <option
          value="created_at"
        >
          Sort by Created Date
        </option>
        <option
          value="updated_at"
        >
          Sort by Updated Date
        </option>
        <option
          value="po_number"
        >
          Sort by PO Number
        </option>
        <option
          value="total_amount"
        >
          Sort by Amount
        </option>
        <option
          value="status"
        >
          Sort by Status
        </option>
      </select>
      <button
        class="sort-button"
        title="Sort Ascending"
      >
        ↓
      </button>
    </div>
  </div>
  <div
    class="empty-state"
  >
    <h3>
      No Purchase Orders Found
    </h3>
    <p>
      Get started by creating your first purchase order.
    </p>
    <button
      class="btn btn-primary"
    >
      Create Your First PO
    </button>
  </div>
</div>
`;

exports[`POList Snapshot Testing should match snapshot with error state 1`] = `
<div
  class="po-list-error"
>
  <div
    class="error-message"
  >
    <h3>
      Error Loading Purchase Orders
    </h3>
    <p>
      Test error message
    </p>
  </div>
</div>
`;

exports[`POList Snapshot Testing should match snapshot with filtered results 1`] = `
<div
  class="po-list"
>
  <div
    class="list-header"
  >
    <div
      class="header-left"
    >
      <h2>
        Purchase Orders
      </h2>
      <div
        class="po-stats"
      >
        <span
          class="stat"
        >
          Total: 
          1
        </span>
        <span
          class="stat"
        >
          Pending: 
          1
        </span>
        <span
          class="stat"
        >
          Approved: 
          0
        </span>
      </div>
    </div>
    <button
      class="btn btn-primary"
    >
      Create New PO
    </button>
  </div>
  <div
    class="list-controls"
  >
    <div
      class="search-bar"
    >
      <input
        class="search-input"
        placeholder="Search by PO number, manufacturer, or distributor..."
        type="text"
        value=""
      />
    </div>
    <div
      class="filters"
    >
      <select
        class="filter-select"
      >
        <option
          value=""
        >
          All Statuses
        </option>
        <option
          value="pending"
        >
          Pending
        </option>
        <option
          value="approved"
        >
          Approved
        </option>
        <option
          value="rejected"
        >
          Rejected
        </option>
        <option
          value="cancelled"
        >
          Cancelled
        </option>
      </select>
      <select
        class="filter-select"
      >
        <option
          value="created_at"
        >
          Sort by Created Date
        </option>
        <option
          value="updated_at"
        >
          Sort by Updated Date
        </option>
        <option
          value="po_number"
        >
          Sort by PO Number
        </option>
        <option
          value="total_amount"
        >
          Sort by Amount
        </option>
        <option
          value="status"
        >
          Sort by Status
        </option>
      </select>
      <button
        class="sort-button"
        title="Sort Ascending"
      >
        ↓
      </button>
    </div>
  </div>
  <div
    class="po-table-container"
  >
    <table
      class="po-table"
    >
      <thead>
        <tr>
          <th
            class="sortable"
          >
            PO Number 
            ↕️
          </th>
          <th>
            Manufacturer
          </th>
          <th>
            Distributor
          </th>
          <th
            class="sortable"
          >
            Amount 
            ↕️
          </th>
          <th
            class="sortable"
          >
            Status 
            ↕️
          </th>
          <th
            class="sortable"
          >
            Created 
            ↓
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="po-row"
        >
          <td
            class="po-number"
          >
            <button
              class="link-button"
            >
              PO-2024-001
            </button>
          </td>
          <td>
            Emerson Process Management
          </td>
          <td>
            Industrial Valve Solutions Inc
          </td>
          <td
            class="amount"
          >
            $1,000.50
          </td>
          <td>
            <span
              class="status-badge status-pending"
            >
              Pending
            </span>
          </td>
          <td
            class="date"
          >
            Jan 15, 2024, 10:30 AM
          </td>
          <td>
            <div
              class="action-buttons"
            >
              <button
                class="btn btn-sm btn-secondary"
                title="View Details"
              >
                View
              </button>
              <button
                class="btn btn-sm btn-success"
                title="Approve PO"
              >
                Approve
              </button>
              <button
                class="btn btn-sm btn-danger"
                title="Reject PO"
              >
                Reject
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`POList Snapshot Testing should match snapshot with loading state 1`] = `
<div
  class="po-list"
>
  <div
    class="list-header"
  >
    <div
      class="header-left"
    >
      <h2>
        Purchase Orders
      </h2>
      <div
        class="po-stats"
      >
        <span
          class="stat"
        >
          Total: 
          3
        </span>
        <span
          class="stat"
        >
          Pending: 
          1
        </span>
        <span
          class="stat"
        >
          Approved: 
          1
        </span>
      </div>
    </div>
    <button
      class="btn btn-primary"
      disabled=""
    >
      Create New PO
    </button>
  </div>
  <div
    class="list-controls"
  >
    <div
      class="search-bar"
    >
      <input
        class="search-input"
        placeholder="Search by PO number, manufacturer, or distributor..."
        type="text"
        value=""
      />
    </div>
    <div
      class="filters"
    >
      <select
        class="filter-select"
      >
        <option
          value=""
        >
          All Statuses
        </option>
        <option
          value="pending"
        >
          Pending
        </option>
        <option
          value="approved"
        >
          Approved
        </option>
        <option
          value="rejected"
        >
          Rejected
        </option>
        <option
          value="cancelled"
        >
          Cancelled
        </option>
      </select>
      <select
        class="filter-select"
      >
        <option
          value="created_at"
        >
          Sort by Created Date
        </option>
        <option
          value="updated_at"
        >
          Sort by Updated Date
        </option>
        <option
          value="po_number"
        >
          Sort by PO Number
        </option>
        <option
          value="total_amount"
        >
          Sort by Amount
        </option>
        <option
          value="status"
        >
          Sort by Status
        </option>
      </select>
      <button
        class="sort-button"
        title="Sort Ascending"
      >
        ↓
      </button>
    </div>
  </div>
  <div
    class="loading-state"
  >
    <div
      class="loading-spinner"
      data-testid="loading-spinner"
    />
    <p>
      Loading purchase orders...
    </p>
  </div>
</div>
`;
